File: python/alignseq_oauth.py
import OAuthSimple
import httplib2
import urllib

class AlignSeq_OAuth(object):

    client = None

    def __init__(self, consumer_key, consumer_secret):
        self.client = OAuthSimple.OAuthSimple(apiKey=consumer_key, sharedSecret=consumer_secret)


    def make_request(self, method, url, opt):
        self.client.reset()

        obj = self.client.sign({'action':method, 'parameters':opt, 'path':url});

        if method != 'POST':
            url = obj.get('signed_url')
            body = None
        else:
            body = urllib.urlencode(opt).replace('+', '%20')


        return httplib2.Http.request(httplib2.Http(timeout=30), url, method=method, body=body, \
                   headers={'Authorization':obj.get('header'), 'Content-Type': 'application/x-www-form-urlencoded'})

File: php/alignseq_oauth.php
<?php

require(__DIR__.'/../oauthsimple/php/OAuthSimple.php');

class AlignSeq_OAuth {

	private $_consumer = NULL;

	public function __construct($consumer_key, $consumer_secret) {
		$this->_consumer = new OAuthSimple($consumer_key, $consumer_secret);
	}

	public function make_request($method, $url, $opt) {
		$method = strtoupper($method);

		$this->_consumer->reset();

		$obj = $this->_consumer->sign(array(
			'action'=> $method,
			'path'=> $url,
			'parameters'=> $opt
		));

		$ch = curl_init();
		curl_setopt_array($ch, array(
			CURLOPT_TIMEOUT => 30,
			CURLOPT_CUSTOMREQUEST => $method,
			CURLOPT_URL => ($method === 'POST') ? $url : $obj['signed_url'],
			CURLOPT_HTTPHEADER => array(
				'Authorization' => $obj['header']
			),
			CURLOPT_RETURNTRANSFER => TRUE,
			CURLOPT_FAILONERROR => FALSE
		));

		if($method === 'POST') {
			curl_setopt($ch, CURLOPT_POSTFIELDS, http_build_query($obj['parameters'], NULL, '&'));
		}

		$r = curl_exec($ch);
		$s = curl_getinfo($ch, CURLINFO_HTTP_CODE);

		curl_close($ch);
		return (object) array('status' => $s, 'body' => $r);
	}

}
